###반복문

131. for문의 실행결과를 예측하라.


In [1]:
과일 = ["사과", "귤", "수박"]
for 변수 in 과일:
 print(변수)


사과
귤
수박


132. for문의 실행결과를 예측하라.


In [2]:
과일 = ["사과", "귤", "수박"]
for 변수 in 과일:
  print("#####") #변수 입력

#####
#####
#####


133. 다음 for 문과 동일한 기능을 수행하는 코드를 작성하세요.
for 변수 in ["A", "B", "C"]:
  print(변수)


In [3]:
변수 = "A"
print(변수)
변수 = "B"
print(변수)
변수 = "C"
print(변수)

A
B
C


134. for문을 풀어서 동일한 동작을하는 코드를 작성하라.

for 변수 in ["A", "B", "C"]:
  print("출력:", 변수)

In [4]:
변수 = "A"
print("출력:", 변수)
변수 = "B"
print("출력:", 변수)
변수 = "C"
print("출력:", 변수)

출력: A
출력: B
출력: C


135. for문을 풀어서 동일한 동작을 하는 코드를 작성하라.

for 변수 in ["A", "B", "C"]:
  b = 변수.lower()
  print("변환:", b)

In [5]:
변수 = "A"
b = 변수.lower()
print("변환:", b)
변수 = "B"
b = 변수.lower()
print("변환:", b)
변수 = "C"
b = 변수.lower()
print("변환:", b)

변환: a
변환: b
변환: c


136. 다음 코드를 for문으로 작성하라.

변수 = 10
print(변수)
변수 = 20
print(변수)
변수 = 30
print(변수)

In [6]:
for 변수 in [10, 20, 30]:
  print(변수)



10
20
30


137. 다음 코드를 for문으로 작성하라.

print(10)
print(20)
print(30)

In [7]:
for 변수 in [10,20,30] :
  print(변수)


10
20
30


138. 다음 코드를 for문으로 작성하라.

print(10)
print("-------")
print(20)
print("-------")
print(30)
print("-------")

In [8]:
for 변수 in [10,20,30] :
  print(변수)
  print("-------")



10
-------
20
-------
30
-------


139. 다음 코드를 for문으로 작성하라.

print("++++")
print(10)
print(20)
print(30)

In [9]:
print("++++")
for 변수 in [10,20,30] :
  print(변수)



++++
10
20
30


140. 다음 코드를 for문으로 작성하라.

print("-------")
print("-------")
print("-------")
print("-------")

In [10]:
for 변수 in [1,2,3,4] :
  print("-------")



-------
-------
-------
-------


141. 다음과 같이 판매가가 저장된 리스트가 있을 때 부가세가 포함된 가격을 for 문을 사용해서 화면에 출력하라. 단 부가세는 10원으로 가정한다.

리스트 = [100, 200, 300]

In [11]:

for 리스트 in [100,200,300] :
  print(리스트+10)

110
210
310


142. for 문을 사용해서 리스트에 저장된 값을 다음과 같이 출력하라.

리스트 = ["김밥", "라면", "튀김"]
오늘의 메뉴: 김밥
오늘의 메뉴: 라면
오늘의 메뉴: 튀김

In [12]:

for 리스트 in ["김밥", "라면", "튀김"] :
  print('오늘의 메뉴:' + 리스트)

오늘의 메뉴:김밥
오늘의 메뉴:라면
오늘의 메뉴:튀김


143. 리스트에 주식 종목이름이 저장돼 있다.

리스트 = ["SK하이닉스", "삼성전자", "LG전자"]
저장된 문자열의 길이를 다음과 같이 출력하라.

6
4
4

In [13]:
리스트 = ["SK하이닉스", "삼성전자", "LG전자"]
for 리스트 in ["SK하이닉스", "삼성전자", "LG전자"] :
  print(len(리스트))


6
4
4


144. 리스트에는 동물이름이 문자열로 저장돼 있다.

리스트 = ['dog', 'cat', 'parrot']
동물 이름과 글자수를 다음과 같이 출력하라.

dog 3
cat 3
parrot 6

In [14]:
리스트 = ['dog', 'cat', 'parrot']
for 이름 in 리스트:
  print(이름, len(이름))


dog 3
cat 3
parrot 6


145. 리스트에 동물 이름 저장돼 있다.

리스트 = ['dog', 'cat', 'parrot']
for문을 사용해서 동물 이름의 첫 글자만 출력하라.

d
c
p

In [15]:
리스트 = ['dog', 'cat', 'parrot']
for 리스트 in ['dog', 'cat', 'parrot'] :
  print(리스트[0])

d
c
p


146. 리스트에는 세 개의 숫자가 바인딩돼 있다.

리스트 = [1, 2, 3]
for문을 사용해서 다음과 같이 출력하라.

3 x 1
3 x 2
3 x 3

In [16]:
리스트 = [1, 2, 3]
for 리스트 in [1,2,3] :
  print("3 x", 리스트)


3 x 1
3 x 2
3 x 3


137. 리스트에는 세 개의 숫자가 바인딩돼 있다.

리스트 = [1, 2, 3]
for문을 사용해서 다음과 같이 출력하라.

3 x 1 = 3
3 x 2 = 6
3 x 3 = 9

In [17]:
리스트 = [1, 2, 3]
for 리스트 in [1,2,3] :
  print("3 x ", 리스트, "=", 3 * 리스트)

3 x  1 = 3
3 x  2 = 6
3 x  3 = 9


148. 리스트에는 네 개의 문자열이 바인딩돼 있다.

리스트 = ["가", "나", "다", "라"]
for문을 사용해서 다음과 같이 출력하라.

나
다
라

In [18]:
리스트 = ["가", "나", "다", "라"]
for 변수 in 리스트[1:]:
  print(변수)

나
다
라


149. 리스트에는 네 개의 문자열이 바인딩돼 있다.

리스트 = ["가", "나", "다", "라"]
for문을 사용해서 다음과 같이 출력하라.

가
다

In [19]:
리스트 = ["가", "나", "다", "라"]
for 변수 in 리스트[: : 2] :
  print(변수)

가
다


150. 리스트에는 네 개의 문자열이 바인딩돼 있다.

리스트 = ["가", "나", "다", "라"]
for문을 사용해서 다음과 같이 출력하라.

라
다
나
가

In [20]:
리스트 = ["가", "나", "다", "라"]
for 변수 in 리스트[::-1]:
 print(변수)

라
다
나
가


151. 리스트에는 네 개의 정수가 저장돼 있다.

리스트 = [3, -20, -3, 44]
for문을 사용해서 리스트의 음수를 출력하라.

-20
-3

In [21]:
리스트 = [3, -20, -3, 44]
for 변수 in 리스트 :
  if 변수 < 0 :
    print(변수)

-20
-3


152. for문을 사용해서 3의 배수만을 출력하라.
3

In [22]:
리스트 = [3, 100, 23, 44]

for 변수 in 리스트 :
    if 변수 % 3 == 0 :
      print(변수)

3


153. 리스트에서 20 보다 작은 3의 배수를 출력하라

리스트 = [13, 21, 12, 14, 30, 18]

In [23]:
리스트 = [13, 21, 12, 14, 30, 18]

for 변수 in 리스트 :
    if 변수 % 3 == 0 and 변수 < 20 :
      print(변수)



12
18


154. 리스트에서 세 글자 이상의 문자를 화면에 출력하라

리스트 = ["I", "study", "python", "language", "!"]



In [24]:
리스트 = ["I", "study", "python", "language", "!"]

for 변수 in 리스트 :
  if len(변수) >= 3:
    print(변수)


study
python
language


155. 리스트에서 대문자만 화면에 출력하라.
리스트 = ["A", "b", "c", "D"]

In [25]:
리스트 = ["A", "b", "c", "D"]

for 변수 in 리스트 :
    if 변수.isupper(): ##
      print(변수)

A
D


156. 리스트에서 소문자만 화면에 출력하라.
리스트 = ["A", "b", "c", "D"]

In [26]:
for 변수 in 리스트 :
    if 변수.islower(): ##
      print(변수)

b
c


157. 이름의 첫 글자를 대문자로 변경해서 출력하라.
리스트 = ['dog', 'cat', 'parrot']

In [27]:
리스트 = ['dog', 'cat', 'parrot']
for 변수 in 리스트 :
  첫글자 = 변수[0]
  대문자 = 첫글자.upper()
  print(대문자 + 변수[1:])



Dog
Cat
Parrot


158. 파일 이름이 저장된 리스트에서 확장자를 제거하고 파일 이름만 화면에 출력하라. (힌트: split() 메서드)
리스트 = ['hello.py', 'ex01.py', 'intro.hwp']

In [28]:
리스트 = ['hello.py', 'ex01.py', 'intro.hwp']
for 변수 in 리스트 :
  splited = 변수.split('.')
  print(splited[0])


hello
ex01
intro


159. 파일 이름이 저장된 리스트에서 확장자가 .h인 파일 이름을 출력하라.



In [29]:
리스트 = ['intra.h', 'intra.c', 'define.h', 'run.py']
for 변수 in 리스트 :
  splited = 변수.split('.')
  if splited[1] == 'h' :
   print(splited[0])



intra
define


160. 파일 이름이 저장된 리스트에서 확장자가 .h나 .c인 파일을 화면에 출력하라.



In [30]:
리스트 = ['intra.h', 'intra.c', 'define.h', 'run.py']
for 변수 in 리스트 :
  splited = 변수.split('.')
  if splited[1] == 'h' or splited[1] == 'c':
   print(splited[0])


intra
intra
define


161. for문과 range 구문을 사용해서 0~99까지 한 라인에 하나씩 순차적으로 출력하는 프로그램을 작성하라.

In [31]:
for 변수 in range(100) :
  print(변수)

0
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99


162, 월드컵은 4년에 한 번 개최된다. range()를 사용하여 2002~2050년까지 중 월드컵이 개최되는 연도를 출력하라.

In [32]:
for 변수 in range(2002, 2050, 4) :
  print(변수)

2002
2006
2010
2014
2018
2022
2026
2030
2034
2038
2042
2046


163. 1부터 30까지의 숫자 중 3의 배수를 출력하라.



In [33]:
for 변수 in range(3, 31, 3) :
  print(변수)

3
6
9
12
15
18
21
24
27
30


164. 99부터 0까지 1씩 감소하는 숫자들을, 한 라인에 하나씩 출력하라.

In [34]:
for 변수 in range(100):
    print(99 - 변수)

99
98
97
96
95
94
93
92
91
90
89
88
87
86
85
84
83
82
81
80
79
78
77
76
75
74
73
72
71
70
69
68
67
66
65
64
63
62
61
60
59
58
57
56
55
54
53
52
51
50
49
48
47
46
45
44
43
42
41
40
39
38
37
36
35
34
33
32
31
30
29
28
27
26
25
24
23
22
21
20
19
18
17
16
15
14
13
12
11
10
9
8
7
6
5
4
3
2
1
0


165. for문을 사용해서 아래와 같이 출력하라.

0.0
0.1
0.2
0.3
0.4
0.5
...
0.9

In [35]:
for num in range(10) :
    print(num / 10)

0.0
0.1
0.2
0.3
0.4
0.5
0.6
0.7
0.8
0.9


166. 구구단 3단을 출력하라.

In [36]:
for i in range(1, 10) : #list of number
 print (3, 'x', i, '=', 3 * i)

3 x 1 = 3
3 x 2 = 6
3 x 3 = 9
3 x 4 = 12
3 x 5 = 15
3 x 6 = 18
3 x 7 = 21
3 x 8 = 24
3 x 9 = 27


167. 구구단 3단을 출력하라. 단 홀수 번째만 출력한다.

In [37]:
for i in range(1, 10) :
   if i % 2 == 1 :
    print (3, 'x', i, '=', 3 * i)

3 x 1 = 3
3 x 3 = 9
3 x 5 = 15
3 x 7 = 21
3 x 9 = 27


168. 1~10까지의 숫자에 대해 모두 더한 값을 출력하는 프로그램을 for 문을 사용하여 작성하라.



In [38]:
num = 0
for i in range(1, 11):
  num += i
print (num)

55


169. 1~10까지의 숫자 중 모든 홀수의 합을 출력하는 프로그램을 for 문을 사용하여 작성하라.



In [39]:
num = 0
for i in range(1, 11, 2):
 num += i #num = num +i

print(num)





25


170. 1~10까지의 숫자를 모두 곱한 값을 출력하는 프로그램을 for 문을 사용하여 작성하라.



In [40]:
num = 1 #
for i in range(1, 11):
 num *= i
print (num)

3628800


171. 아래와 같이 리스트의 데이터를 출력하라. 단, for문과 range문을 사용하라.



In [41]:
price_list = [32100, 32150, 32000, 32500]

for i in range(len(price_list)):
    print(price_list[i])

32100
32150
32000
32500


172. 아래와 같이 리스트의 데이터를 출력하라. 단, for문과 range문을 사용하라.



In [42]:
price_list = [32100, 32150, 32000, 32500]

for i in range(0,4):
  print(price_list[i])

32100
32150
32000
32500


173. 아래와 같이 리스트의 데이터를 출력하라. 단, for문과 range문을 사용하라.



In [43]:
price_list = [32100, 32150, 32000, 32500]



174. 아래와 같이 리스트의 데이터를 출력하라. 단, for문과 range문을 사용하라.
100 32150
110 32000
120 32500

In [44]:

price_list = [32100, 32150, 32000, 32500]



175. my_list를 아래와 같이 출력하라.

가 나
나 다
다 라

In [45]:

my_list = ["가", "나", "다", "라"]

176. 리스트를 아래와 같이 출력하라.


가 나 다
나 다 라
다 라 마

In [46]:
my_list = ["가", "나", "다", "라", "마"]

177. 반복문과 range 함수를 사용해서 my_list를 아래와 같이 출력하라.


라 다
다 나
나 가

In [47]:
my_list = ["가", "나", "다", "라"]

178. 리스트에는 네 개의 정수가 저장되어 있다. 각각의 데이터에 대해서 자신과 우측값과의 차분값을 화면에 출력하라.

100
200
400

In [48]:
my_list = [100, 200, 400, 800]

179. 리스트에는 6일 간의 종가 데이터가 저장되어 있다. 종가 데이터의 3일 이동 평균을 계산하고 이를 화면에 출력하라.


첫 번째 줄에는 100, 200, 400의 평균값이 출력된다. 두 번째 줄에는 200, 400, 800의 평균값이 출력된다. 같은 방식으로 나머지 데이터의 평균을 출력한다.

In [49]:
my_list = [100, 200, 400, 800, 1000, 1300]

180. 리스트에 5일간의 저가, 고가 정보가 저장돼 있다. 고가와 저가의 차를 변동폭이라고 정의할 때, low, high 두 개의 리스트를 사용해서 5일간의 변동폭을 volatility 리스트에 저장하라.

In [50]:
low_prices  = [100, 200, 400, 800, 1000]
high_prices = [150, 300, 430, 880, 1000]